railties/lib/rails/generators/rails/app/templates/config/initializers/new_framework_defaults_7_0.rb.tt

```diff
@@ -83,11 +83,13 @@
 # Rails.application.config.active_storage.variant_processor = :vips
 
 # If you're upgrading and haven't set `cookies_serializer` previously, your cookie serializer
-# was `:marshal`. Convert all cookies to JSON, using the `:hybrid` formatter.
+# was `:marshal`. The default for new apps is `:json`.
 #
-# If you're confident all your cookies are JSON formatted, you can switch to the `:json` formatter.
+# You can convert all cookies to JSON using the `:hybrid` formatter. It is fine to use
+#`:hybrid` long term; you should do that unless you're confident that *all* your cookies
+# have been converted to JSON.
 #
-# Continue to use `:marshal` for backward-compatibility with old cookies.
+# You can also use `:marshal` for backward-compatibility with old cookies.
```

Comment:
I think would should recommend against using `:marshal` as it's a possible security risk.
The only reason to keep using it if you store Ruby objects that can't be serialized to json like `Date`, but that isn't best practice as mentioned in https://guides.rubyonrails.org/action_controller_overview.html#cookies.

```suggestion
# If your cookies can't yet be serialized to JSON continue using `:marshal` for backward-compatibility.
```

We can also link to the cookies guide for more information: https://guides.rubyonrails.org/action_controller_overview.html#cookies.